api/layer/object.go

```diff
@@ -88,11 +88,7 @@ func (n *layer) objectSearch(ctx context.Context, p *findParams) ([]*object.ID,
 	opts.AddRootFilter()
 
 	for _, filter := range p.filters {
-		if val, err := url.QueryUnescape(filter.val); err != nil {
-			return nil, err
-		} else if val != "" {
-			opts.AddFilter(filter.attr, val, object.MatchStringEqual)
-		}
+		opts.AddFilter(filter.attr, filter.val, object.MatchStringEqual)
 	}
 
 	if prefix, err := url.QueryUnescape(p.prefix); err != nil {
```

Comment:
Why we left `QueryUnescape` for prefix? It's provided as url parameter like object name